apps/intellij/src/main/kotlin/dev/nx/console/run/NxRunConfigurationProducer.kt

```diff
@@ -0,0 +1,145 @@
+package dev.nx.console.run
+
+import com.intellij.execution.actions.ConfigurationContext
+import com.intellij.execution.actions.LazyRunConfigurationProducer
+import com.intellij.execution.configurations.ConfigurationFactory
+import com.intellij.json.psi.JsonFile
+import com.intellij.json.psi.JsonObject
+import com.intellij.json.psi.JsonProperty
+import com.intellij.json.psi.JsonStringLiteral
+import com.intellij.openapi.util.Ref
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.util.parentOfType
+
+data class NxRunSettings(
+    val nxProjects: String = "",
+    val nxTargets: String = "",
+    val arguments: String = ""
+)
+
+class NxRunConfigurationProducer : LazyRunConfigurationProducer<NxCommandConfiguration>() {
+    override fun getConfigurationFactory(): ConfigurationFactory =
+        NxCommandConfigurationType.getInstance()
+
+    override fun setupConfigurationFromContext(
+        configuration: NxCommandConfiguration,
+        context: ConfigurationContext,
+        sourceElement: Ref<PsiElement>
+    ): Boolean {
+
+        val runSettings =
+            createRunSettingsFromContext(
+                NxRunSettings(
+                    configuration.nxProjects,
+                    configuration.nxTargets,
+                    configuration.arguments
+                ),
+                context,
+                sourceElement
+            )
+
+        if (runSettings == null) {
+            return false
+        } else {
+            setupConfigurationFromSettings(configuration, runSettings)
+            return true
+        }
+    }
+
+    private fun setupConfigurationFromSettings(
+        configuration: NxCommandConfiguration,
+        runSettings: NxRunSettings
+    ) {
+        // configuration.setRunSettings(runSettings)
+        configuration.nxProjects = runSettings.nxProjects
+        configuration.nxTargets = runSettings.nxTargets
+        configuration.arguments = runSettings.arguments
+
+        // configuration.name = "${runSettings.nxProjects}[${runSettings.nxTargets}]"
+        configuration.setGeneratedName()
+    }
+
+    private fun createRunSettingsFromContext(
+        runSettings: NxRunSettings,
+        context: ConfigurationContext,
+        sourceElement: Ref<PsiElement>?
+    ): NxRunSettings? {
+        val element = getElement(context)
+        if (element != null && element.isValid) {
+            val psiProjectJsonFile = getContainingProjectJsonFile(element)
```

Comment:
Do you think there's a way you could refactor this code a bit? Maybe by having some early return conditions instead of nesting conditionals deeper and deeper? I'm having some trouble reading the 4th nested `else` and I think folks in the future will too.

Reply:
I mimic the behaviour of NpmRunConfigurationProducer, the code is similar but sure it can be reworked